simulator/source/Combat_simulator.cpp

```diff
@@ -604,15 +604,14 @@ bool Combat_simulator::start_cast_slam(bool mh_swing, double rage, double& swing
     }
     if (use_sl)
     {
-        if (swing_time_left > config.combat.slam_spam_max_time)
+        if (mh_swing || swing_time_left > config.combat.slam_spam_max_time)
```

Comment:
on a mh_swing, swing_time_left used to already be reset (to swing_timer / haste) here, as part of calling swing_weapon(). now it's <= 0, since the reset happens "at end of turn"